ArduSub/ArduSub.cpp

```diff
@@ -345,4 +345,40 @@ bool Sub::control_check_barometer()
     return true;
 }
 
+/*
+  get the current wp index
+  return false if failed or n/a
+ */
+bool Sub::get_wp_index(uint16_t &index)
```

Comment:
We `send_nav_controller_output` for Sub - can't see why we shouldn't have implementations here.

Reply:
by playing with the code it seems that a simple

uint16_t get_wp_index() is way simpler to handle than bool get_wp_index(uint16_t &index).
All methods called by this function are already designed to either return 0 or the correct value, there's no handling of a true/false result

Reply:
And if `AP_Mission` isn't compiled in?

0 is a valid index.